app/pages/settings/change-password-form.jsx

```diff
@@ -0,0 +1,147 @@
+import React from 'react';
+import auth from 'panoptes-client/lib/auth';
+import counterpart from 'counterpart';
+
+const MIN_PASSWORD_LENGTH = 8;
+
+export default class ChangePasswordForm extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      old: '',
+      new: '',
+      confirmation: '',
+      inProgress: false,
+      success: false,
+      error: null
+    };
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.changePasswordForm = React.createRef();
+  }
+
+  handleSubmit(e) {
+    e.preventDefault();
+    const current = this.state.old;
+    const replacement = this.state.new;
+
+    this.setState({
+      inProgress: true,
+      success: false,
+      error: null
+    });
+
+    auth.changePassword({ current, replacement })
+      .then(() => {
+        this.setState({ success: true });
+        this.changePasswordForm.current.reset();
+      })
+      .catch((error) => {
+        this.setState({ error });
+      })
+      .then(() => {
+        this.setState({ inProgress: false });
+      });
+  }
+
+  doesntMatch() {
+    return this.state.new !== this.state.confirmation;
+  }
+
+  tooShort() {
+    return this.state.new.length < MIN_PASSWORD_LENGTH;
+  }
+
+  renderStatus() {
+    let status = null;
+    if (this.state.inProgress) {
+      status = <i className="fa fa-spinner fa-spin form-help"></i>;
+    } else if (this.state.success) {
+      status = <i className="fa fa-check-circle form-help success"></i>;
+    } else if (this.state.error) {
+      status = <small className="form-help error">{this.state.error.toString()}</small>;
+    }
+    return status;
+  }
+
+  render() {
+    return (
+      <form ref={this.changePasswordForm} method="POST" onSubmit={this.handleSubmit}>
+        <p>
+          <strong>{counterpart('userSettings.account.changePassword.heading')}</strong>
+        </p>
+        <table className="standard-table">
```

Comment:
This is a comment for a later PR since the original component did this. It's not great for accessibility to use a table for layout purposes rather than tabular data. You can use CSS to make the markup render like a table with `display: table;` instead and use good semantic HTML for a form. MDN has a good list of what elements should be used with a form: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/form. The `label` tags should be wrapping the `input`s.